src/app/services/nodeRewardsProgram.service.js

```diff
@@ -0,0 +1,173 @@
+
+        // Set no mosaics
+        transferTransaction.mosaics = null;
+
+        return nem.model.transactions.prepare('transferTransaction')(this.common, transferTransaction, this._Wallet.network);
+    }
+
+    /**
+     * Checks if an address is the current enrollment address.
+     *
+     * @param {string} address - Address to check.
+     *
+     * @return {Promise<boolean>} - If the address to check matches the current enrollment address.
+     */
+    async checkEnrollmentAddress(address) {
+        const response = await fetch(this.apiBaseUrl + `/enrollment/check/address/${address}`);
+
+        if (!response.ok) {
+            throw Error('failed_to_validate_enroll_address');
+        }
+        
+        const status = await response.text();
+        
+        return status === 'true';
+    }
+
+    /**
+     * Checks if a signer public key is enrolled in the current period.
+     *
+     * @param {string} publicKey - Delegated harvesting public key to check.
+     *
+     * @return {Promise<boolean>} - If the public key is enrolled in the current period.
+     */
+    async checkEnrollmentStatus(publicKey) {
+        const successEnrollmentResponse = await fetch(this.apiBaseUrl + `/enrollment/successes/${publicKey}?count=1`);
+
+        if (!successEnrollmentResponse.ok) {
+            throw Error('failed_to_get_success_enrollments');
+        }
+
+        const successEnrollments = await successEnrollmentResponse.json();
+
+        if (successEnrollments.length === 0) {
+            return false;
+        }
+
+        const latestSuccessEnrollment = successEnrollments[0];
+        const enrollAddress = latestSuccessEnrollment.recipientAddress;
+
+        return this.checkEnrollmentAddress(enrollAddress);
+    }
+
+    /**
+     * Gets codeword dependent hash for the current period given a public key.
+     *
+     * @param {string} publicKey - Signer public key to use in hash calculation.
+     *
+     * @return {Promise<string>} - The result of keccak_256(codeword || public_key) as a hex string, which should be used in enrollment messages.
+     */
+    async getCodewordHash(publicKey) {
+        const response = await fetch(this.apiBaseUrl + `/codeword/${publicKey}`);
+
+        if (!response.ok) {
+            throw Error('failed_to_get_codeword_hash');
+        }
+        
+        return await response.text();
+    }
+
+    /**
+     * Gets payout information for a single node.
+     *
+     * @param {string} nodeId - Id of the node to search.
+     * @param {number} pageNumber - Page number to return.
+     *
+     * @return {Promise<Array<object>>} - List of payouts.
+     */
+    async getNodePayouts(nodeId, pageNumber) {
+        const count = 15;
+        const offset = count * pageNumber;
+
+        const response = await fetch(this.apiBaseUrl + `/node/${nodeId}/payouts?count=${count}&offset=${offset}`);
+
+        if (!response.ok) {
+            throw Error('failed_to_get_payouts_page');
+        }
```

Comment:
you should be able to refactor this block

const json = await this.get(`/node/${nodeId}/payouts?count=${count}&offset=${offset}`, `failed_to_get_payouts_page`)

since pattern is same for all GETs

Reply:
Refactored.